packages/stardog-graphql-language-server/src/GraphQlLanguageServer.ts

```diff
@@ -192,6 +207,131 @@ export class GraphQlLanguageServer extends AbstractLanguageServer<
+      (directive) =>
+        isTopLevel
+          ? is(directive, stardogGraphQlTokenMap.TopLevel.name)
+          : !is(directive, stardogGraphQlTokenMap.TopLevel.name)
+    );
+
+    stardogSpecificDirectivesForLevel.forEach((stardogSpecificDirective) =>
+      candidates.push({
+        ...directiveCandidate,
+        nextTokenType: stardogSpecificDirective,
+      })
+    );
+  }
+
+  private addStardogSpecificArgumentsToCompletionCandidates(
+    candidates: CompletionCandidate[],
+    tokenVectorForContentAssist: IToken[]
+  ) {
+    // Find the candidate, if any, that is at the spot where an Argument Alias
+    // is valid.
+    const argumentAliasCandidate = candidates.find(({ ruleStack }) => {
+      const ruleStackLength = ruleStack.length;
+      return (
+        ruleStack[ruleStackLength - 1] === 'Alias' &&
+        ruleStack[ruleStackLength - 2] === 'Argument'
+      );
+    });
+
+    if (!argumentAliasCandidate) {
+      return;
+    }
+
+    const isDirectiveArgument =
+      argumentAliasCandidate.ruleStack.slice(-4).join('/') ===
+      'Directive/Arguments/Argument/Alias';
+
+    if (isDirectiveArgument) {
+      let containingStardogDirective: IToken;
+      let parensStackCount = 1;
+
+      // Walk back through the tokens until we find the Stardog directive for
+      // which the currently located candidate is an argument (using parens
+      // matching).
+      for (
+        let i = tokenVectorForContentAssist.length - 1;
+        i >= 0 || parensStackCount < 0;
+        i--
+      ) {
+        const currentToken = tokenVectorForContentAssist[i];
+        if (
+          currentToken.tokenType.name === stardogGraphQlTokenMap.LParen.name
+        ) {
+          parensStackCount--;
+        } else if (
+          currentToken.tokenType.name === stardogGraphQlTokenMap.RParen.name
+        ) {
+          parensStackCount++;
+        }
+        if (
+          is(
+            currentToken.tokenType,
+            stardogGraphQlTokenMap.StardogDirective.name
+          ) &&
+          parensStackCount === 0
+        ) {
+          // When we have no extraneous parentheses and we've found a Stardog
+          // directive, we know we've got the Stardog-specific directive for
+          // which the candidate is an argument.
+          containingStardogDirective = currentToken;
+          break;
+        }
+      }
+
+      if (!containingStardogDirective) {
+        return;
+      }
+
+      // Get the allowed argument aliases for the located Stardog-specific
+      // directive, and add them to the completion candidates.
+      const argumentAliasTokenTypes = graphQlUtils.getArgumentTokenTypesForDirectiveNameToken(
+        containingStardogDirective
+      );
+      argumentAliasTokenTypes.forEach((argumentAliasTokenType) =>
+        candidates.push({
+          ...argumentAliasCandidate,
+          nextTokenType: argumentAliasTokenType,
+        })
+      );
```

Comment:
Technially, this is _syntactically_ correct. We could add additional intelligence to `millan`'s GraphQL visitor to narrow things down in these cases, but I was thinking that could be a future ticket rather than a requirement for this (especially because this blocks the "Run Current Query" work in Studio).